PG TileJSON changes, add vector_layers

nyurik opened the pull request:
* make tilejson's `name` be the same as the ID of the source (even if aliased)
* `/catalog` will always show ID, but now it will hide the `name` if it is the same as the `id`
* make `description` be the longer version, e.g. `public.table.column` format - not guaranteed to be stable
* make `vector_layers` have the fields auto-discovered in the PG table
* preserve the order of the serialized json fields

Fixes #583

nyurik committed:
Add vector_layers to PG tilejson from tables

nyurik committed:
catalog should only show name if diff

nyurik committed:
updated readme

nyurik committed:
order json output

stepankuzmin commented on src/srv/server.rs:
```diff
@@ -170,7 +170,7 @@ async fn get_catalog(state: Data<AppState>) -> impl Responder {
                 content_type: info.format.content_type().to_string(),
                 content_encoding: info.encoding.content_encoding().map(ToString::to_string),
-                name: tilejson.name,
+                name: tilejson.name.filter(|v| v != id),
```
What's that filter for?

nyurik replied:
hides the `name` from the output if it is identical to the ID

stepankuzmin reviewed (approved):
LGTM 👍

nyurik committed:
0.7.2 and test fix

We had two metadata fields in the mbtiles metadata that were not
officially supported by tilejson, so removed the generator to keep
ordering simpler.
